hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +48,238 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
     // sync the embedded group
     m_cl->setEmbeddedGroup(m_embed_group);
 
+    // check for collision warnings
+    checkCollisionWarnings(timestep);
+
     // set random grid shift
     m_cl->drawGridShift(timestep);
 
     // update cell list
     m_cl->compute(timestep);
 
+    // create auxillary array for rigid bodies
+    if (m_embed_group)
+        {
+        // resize initial velocity array
+        unsigned int N_tot = m_embed_group->getNumMembers();
+        if (N_tot > m_initial_velocity.getNumElements())
+            {
+            GPUArray<Scalar4> initial_velocity(N_tot, m_exec_conf);
+            m_initial_velocity.swap(initial_velocity);
+            }
+        unsigned int Np_tot = m_pdata->getN();
+        if (Np_tot > m_linmom_change.getNumElements())
+            {
+            GPUArray<Scalar3> linmom_change(Np_tot, m_exec_conf);
+            m_linmom_change.swap(linmom_change);
+            GPUArray<Scalar4> angmom_change(Np_tot, m_exec_conf);
+            m_angmom_change.swap(angmom_change);
+            }
+        beginRigidBodyCollision(timestep);
+        }
+
+    // apply collisions
     rule(timestep);
+
+    // apply collisions to rigid bodies
+    if (m_embed_group)
+        {
+        finishRigidBodyCollision(timestep);
+        }
+    }
+
+void mpcd::CollisionMethod::checkCollisionWarnings(uint64_t timestep)
+    {
+    if (m_checked_collision_warnings)
+        {
+        return;
+        }
+    if (m_embed_group)
```

Comment:
Use white space to make code more readable
```suggestion

    if (m_embed_group)
```